jms/src/main/scala/akka/stream/alpakka/jms/JmsProducerStage.scala

```diff
@@ -21,11 +28,14 @@ private[jms] final class JmsProducerStage[A <: JmsMessage](settings: JmsProducer
   override protected def initialAttributes: Attributes =
     ActorAttributes.dispatcher("akka.stream.default-blocking-io-dispatcher")
 
-  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new GraphStageLogic(shape) with JmsConnector {
 
-      private var jmsProducer: MessageProducer = _
-      private var jmsSession: JmsSession = _
+      private lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+
+      private val jmsProducers: Buffer[JmsMessageProducer] = Buffer(settings.sessionCount, settings.sessionCount)
+
+      private val buffer: Buffer[Holder[A]] = Buffer(settings.sessionCount, settings.sessionCount)
```

Comment:
Would `producerWithMessage` describe better what it holds?

Reply:
how about `inFlightMessagesWithProducer`?